phoenix-core/src/main/java/org/apache/phoenix/mapreduce/FindViewCorruptionTool.java

```diff
@@ -0,0 +1,328 @@
+                throw new IllegalStateException(TABLE_OPTION.getLongOpt() +
+                        " requires an argument.");
+            } else {
+                this.tableName = cmdLine.getOptionValue(TABLE_OPTION.getOpt());
+            }
+        }
+        if (cmdLine.hasOption(GET_CORRUPTED_VIEWS_COUNT_OPTION.getOpt())) {
+            getCorruptedViewCount = true;
+        }
+        outputPath = cmdLine.getOptionValue(OUTPUT_PATH_OPTION.getOpt());
+    }
+
+    private Options getOptions() {
+        final Options options = new Options();
+        options.addOption(OUTPUT_PATH_OPTION);
+        options.addOption(TENANT_ID_OPTION);
+        options.addOption(SCHEMA_OPTION);
+        options.addOption(TABLE_OPTION);
+        options.addOption(GET_CORRUPTED_VIEWS_COUNT_OPTION);
+        options.addOption(HELP_OPTION);
+        return options;
+    }
+
+    private void printHelpAndExit(String errorMessage, Options options) {
+        System.err.println(errorMessage);
+        printHelpAndExit(options, 1);
+    }
+
+    private void printHelpAndExit(Options options, int exitCode) {
+        HelpFormatter formatter = new HelpFormatter();
+        formatter.printHelp("help", options);
+        System.exit(exitCode);
+    }
+
+    private class View {
+        private String tenantId;
+        private String schemaName;
+        private String tableName;
+        private String selectColumnCount;
+        private String columnCount;
+
+        public View (String tenantId, String schemaName, String tableName, String columnCount) {
+            this.schemaName = schemaName == null ? "" : schemaName;
+            this.tableName = tableName  == null ? "" : tableName;
+            this.tenantId = tenantId  == null ? "" : tenantId;
+            this.columnCount = columnCount  == null ? "0" : columnCount;
+            this.selectColumnCount = "0";
+        }
+
+        public String getTenantId() {
+            return this.tenantId;
+        }
+
+        public String getSchemaName() {
+            return this.schemaName;
+        }
+
+        public String getTableName() {
+            return this.tableName;
+        }
+
+        public String getColumnCount() {
+            return this.columnCount;
+        }
+
+        public String getSelectColumnCount() {
+            return this.selectColumnCount;
+        }
+
+        public void setSelectColumnCount(String selectColumnCount) {
+            this.selectColumnCount = selectColumnCount == null ? "0" : selectColumnCount;
+        }
+    }
+
+    private void closeConnection(Connection connection) {
+        try {
+            if (connection != null) {
+                connection.close();
+            }
+        } catch (SQLException e) {
+            LOGGER.error("Failed to close connection: ", e);
+            throw new RuntimeException("Failed to close connection with exception: ", e);
+        }
+    }
+
+    private void writeCorruptedViews() throws IOException {
```

Comment:
Would be good to split the logic so that the actual checker logic is a class that takes in normal arguments (not arg parsing) and returns objects from an API, and then a tool class that does arg parsing and invokes the API. That will make the logic composable (callable from different tools, such as a future PHOENIX-3165) and make both the business logic and the I/O functions easier to test.